libraries/AP_Notify/Buzzer.cpp

```diff
@@ -68,15 +68,57 @@ void Buzzer::update()
 
 void Buzzer::update_pattern_to_play()
 {
+    if (!_flags.initialize_done) {
+        // play a quick beep at startup
+        if (!_flags.startup_beep) {
+            _flags.startup_beep = true;
+            play_pattern(SINGLE_BUZZ);
+            return;
+        }
+        // play beeps when system initialization complete
+        if (!_flags.initialize_started) {
+            if (AP_Notify::flags.initialising) {
+                _flags.initialize_started = true;
+            }
+        } else if (!AP_Notify::flags.initialising) {
+            _flags.initialize_done = true;
+            play_pattern(INITIALIZE_BUZZ);
+            return;
+        }
+    }
+
+    // check if radio failsafe status has changed
+    if (_flags.failsafe_radio != AP_Notify::flags.failsafe_radio) {
+        _flags.failsafe_radio = AP_Notify::flags.failsafe_radio;
+        // play beeps to indicate radio status (if initialization completed)
+        if (_flags.initialize_done) {
+            play_pattern(_flags.failsafe_radio ? RADIOLOST_BUZZ : RADIOBACK_BUZZ);
+            return;
+        }
+    }
+
     // check for arming failed event
     if (AP_Notify::events.arming_failed) {
         // arming failed buzz
         play_pattern(SINGLE_BUZZ);
         return;
     }
 
-    if (AP_HAL::millis() - _pattern_start_time < _pattern_start_interval_time_ms) {
-        // do not interrupt playing patterns / enforce minumum separation
+    // do not interrupt playing patterns
+    if (is_playing_pattern()) {
+        return;
+    }
+
+    // if battery failsafe constantly buzz
+    if (AP_Notify::flags.failsafe_battery) {
+        play_pattern(CONSTANT_BUZZ);
+        return;
+    }
+
+    // if radio failsafe active and system was (ever) armed then
+    // repeat SOS beeps until radio recovered (for lost model locator)
+    if (_flags.failsafe_radio && _flags.was_armed) {
```

Comment:
There's a Notify flag to check for playing the lost vehicle alarm.  There's even a block down there that attempts to handle that condition.

If you think the lost-vehicle condition should be set in the case that we've ever armed and now can't see our RC - then that's a patch to different parts of ArduPilot which decides that a vehicle is "lost".  That way it's not just those with buzzers that benefit, but those using ToneAlarm-enabled devices (and, notionally, other means of letting the device be found (making sounds with the motors?)).

Reply:
I don't disagree, but more-global lost-vehicle condition is not something I have time to try to implement at this point.